compatibility/src/membership.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::BTreeSet;
+use rand::{prelude::ThreadRng, Rng};
+
+type NodeId = u64;
+
+#[derive(
+  bincode_2::Encode, bincode_2::Decode, serde::Serialize, serde::Deserialize, Debug, PartialEq,
+)]
+#[bincode(crate = "bincode_2")]
+pub struct Membership {
+    /// learners set
+    learners: BTreeSet<NodeId>,
+
+    /// Multi configs.
+    configs: Vec<BTreeSet<NodeId>>,
+
+    /// Cache of all node ids.
+    all_members: BTreeSet<NodeId>,
+}
+
+#[test]
+pub fn test() {
+    let mut rng = rand::thread_rng();
+    for _ in 0..1000 {
+        crate::test_same(Membership {
+            learners: random_btreeset(&mut rng, 1000),
+            configs: vec_random_btreeset(&mut rng,),
+            all_members: random_btreeset(&mut rng, 1000),
+        });
+    }
+}
+
+fn vec_random_btreeset(rng: &mut ThreadRng) -> Vec<BTreeSet<NodeId>> {
+  let mut vec = Vec::with_capacity(5);
+  for _ in 0..10 {
```

Comment:
How many cases do you have in your project? I"m wondering if we should add a random amount of btreeset's, between 0 (inclusive) and 100 maybe

Reply:
It's a very low number of entries but I will modify that to use a random amount.